persistence/src/main/scala/com/evolutiongaming/akkaeffect/persistence/EventSourcedStoreOf.scala

```diff
@@ -0,0 +1,167 @@
+package com.evolutiongaming.akkaeffect.persistence
+
+import akka.persistence.SnapshotSelectionCriteria
+import cats.syntax.all._
+import cats.effect.{Async, Ref, Resource, Sync}
+import akka.persistence.journal.AsyncRecovery
+import akka.persistence.snapshot.SnapshotStore
+import com.evolution.akkaeffect.eventsopircing.persistence.{
+  Event,
+  EventSourcedStore,
+  Recovery,
+  Snapshot
+}
+import com.evolutiongaming.catshelper.FromFuture
+import com.evolutiongaming.catshelper.ToTry
+import com.evolutiongaming.sstream.Stream
+
+import java.time.Instant
+import scala.concurrent.Future
+import scala.util.Try
+import com.evolutiongaming.sstream.FoldWhile._
+
+object EventSourcedStoreOf {
+
+  /**
+    * [[EventSourcedStore]] implementation based on Akka Persistence API.
+    *
+    * The implementation delegates snapshot and events load to [[SnapshotStore]] and [[AsyncRecovery]].
+    * Snapshot loaded on [[EventSourcedStore#recover]] F while events loaded lazily:
+    * first events will be available for [[Stream#foldWhileM]] while tail still loaded by [[AsyncRecovery]]
+    *
+    * @param snapshotStore Akka Persistence snapshot (plugin)
+    * @param journal Akka Persistence journal (plugin)
+    * @tparam F effect
+    * @tparam S snapshot
+    * @tparam E event
+    * @return resource of [[EventSourcedStore]]
+    */
+  def fromAkka[F[_]: Async: ToTry, S, E](snapshotStore: SnapshotStore,
```

Comment:
overall I'm more interested in actual implementation that meant to replace `PersistentActor` :)

Reply:
Please take a look on basic part of persistent actor replacement: `com.evolutiongaming.akkaeffect.persistence.EventSourcedActorOf`